libs/foundry-dev-tools/src/foundry_dev_tools/clients/build2.py

```diff
@@ -237,3 +237,62 @@ def submit_dataset_build(
         return self.api_submit_build(
             job_spec_selections=datasets_job_spec_selections, branch=branch, force_build=force_build
         ).json()
+
+    def api_get_jobspec_for_dataset(
+        self,
+        dataset_rid: api_types.DatasetRid,
+        branch: api_types.DatasetBranch = "master",
+        branch_fallbacks: set[api_types.DatasetBranch] | None = None,
+        **kwargs,
+    ) -> dict | None:
```

Comment:
`api_`  methods return requests.Response objects. If you want to do higher level logic, please add a non `_api` method.